orbit-2-rxjava2/src/main/java/com/babylon/orbit2/rxjava2/RxJava2DslPlugin.kt

```diff
@@ -48,29 +49,45 @@ object RxJava2DslPlugin : OrbitDslPlugin {
     ): Flow<Any?> {
         return when (operator) {
             is RxJava2Observable<*, *, *> -> flow.flatMapConcat {
+                if (operator.registerIdling) containerContext.settings.idlingRegistry.increment()
+
                 with(operator as RxJava2Observable<S, E, Any>) {
                     createContext(it).block()
-                }.asFlow().flowOn(containerContext.backgroundDispatcher)
+                }.asFlow().flowOn(containerContext.backgroundDispatcher).also {
+                    if (operator.registerIdling) containerContext.settings.idlingRegistry.decrement()
+                }
```

Comment:
Ditto not sure it will work with an observable this way

We need something like

``` kotlin
createContext(it).block()
.doOnSubscribe {
if (operator.registerIdling) containerContext.settings.idlingRegistry.increment()
}
.doOnComplete {
if (operator.registerIdling) containerContext.settings.idlingRegistry.decrement()
}

```

Reply:
And actually the same applies to all RxJava operators, they're just different flavors of observable in the end